File: backends/ruby-hanami/app/domain/client/value_objects.rb
module Domain
  module Client
    ListItem = Struct.new(:id, :name, :status, :start_at, :stop_at, :created_at, :updated_at, keyword_init: true)

    DetailVo = Struct.new(
      :id, :name, :identifier,
      :post_code, :pref, :city, :address, :building, :tel, :email,
      :status, :start_at, :stop_at, :created_at, :updated_at,
      keyword_init: true
    )
  end
end

File: backends/go-echo/internal/domain/client/value_objects.go
package client

import "time"

type ListItem struct {
	ID        uint64
	Name      string
	Status    int
	StartAt   *time.Time
	StopAt    *time.Time
	CreatedAt time.Time
	UpdatedAt time.Time
}

type DetailVo struct {
	ID        uint64
	Name      string
	Identifier string
	PostCode  string
	Pref      string
	City      string
	Address   string
	Building  string
	Tel       string
	Email     string
	Status    int
	StartAt   *time.Time
	StopAt    *time.Time
	CreatedAt time.Time
	UpdatedAt time.Time
}

type StoreVo struct {
	ID          uint64
	Name        string
	Identifier  string
	Email       string
	AccessToken string
}

type QrVo struct {
	Identifier  string
	DeeplinkURL string
}

type InfoVo struct {
	Identifier string
	Name       string
	Status     int
}

type StartVo struct {
	AccessToken string
}
